### PR TITLE
(PC-23921) feat(profile): add new CreditProgressBar

### PR DESCRIPTION
**Pull request by Lucasbeneston**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-23821

## Checklist

I have:

- [x] Made sure my feature is working on the relevant real / virtual devices (native and web).
- [ ] Written **unit tests** native (and web when implementation is different) for my feature.
- [ ] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]

## Screenshots

**delete** *if no UI change*

| Platform         | Before | After |
| :--------------- | :----: | :---: |
| iOS              |        |       |
| Android          |        |       |
| Phone - Chrome   |        |  Capture d’écran 2023-08-24 à 17 22 54 Capture d’écran 2023-08-24 à 17 22 44 Capture d’écran 2023-08-24 à 17 22 36 Capture d’écran 2023-08-24 à 17 22 28 Capture d’écran 2023-08-24 à 17 21 51 |
| Tablet - Chrome  |        |       |
| Desktop - Chrome |        |       |
| Desktop - Safari |        |       |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a

### CODE REVIEW

**Comment by Lucasbeneston** [comment]
> SonarCloud Quality Gate failed.    [Quality Gate failed](https://sonarcloud.io/dashboard?id=pass-culture_pass-culture-app-native&pullRequest=5204)
> 
> [Bug](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=BUG) [Vulnerability](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=VULNERABILITY) [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=SECURITY_HOTSPOT) [Code Smell](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=CODE_SMELL) [0 Code Smells](https://sonarcloud.io/project/issues?id=pass-culture_pass-culture-app-native&pullRequest=5204&resolved=false&types=CODE_SMELL)
> 
> [](https://sonarcloud.io/component_measures?id=pass-culture_pass-culture-app-native&pullRequest=5204&metric=new_coverage&view=list) [56.5% Coverage](https://sonarcloud.io/component_measures?id=pass-culture_pass-culture-app-native&pullRequest=5204&metric=new_coverage&view=list) [](https://sonarcloud.io/component_measures?id=pass-culture_pass-culture-app-native&pullRequest=5204&metric=new_duplicated_lines_density&view=list) [0.0% Duplication](https://sonarcloud.io/component_measures?id=pass-culture_pass-culture-app-native&pullRequest=5204&metric=new_duplicated_lines_density&view=list)
> 
> warning The version of Java (11.0.11) you have used to run this analysis is deprecated and we will stop accepting it soon. Please update to at least Java 17. Read more [here](https://docs.sonarcloud.io/appendices/scanner-environment/)
> 
> idea Catch issues before they fail your Quality Gate with our IDE extension sonarlint [SonarLint](https://www.sonarsource.com/products/sonarlint/features/connected-mode/?referrer=sonarcloud-welcome)

Normal car j'ai supprimé des tests / snapshots

File: src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { CreditProgressBar } from 'features/profile/components/CreditInfo/CreditProgressBar'
+import React from 'react'
+import { View } from 'react-native'
+import styled from 'styled-components/native'
+
+import { getSpacing } from 'ui/theme'
+
+interface Props {
+  progress: '1/3' | '2/3' | '3/3' | '1/2' | '2/2'
+}
```
Comment (marineshaw) [inline src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx:10]: Je verrais plus deux props avec CurrentStep et TotalStep (surtout que tu différencies les deux derrière) tu en penses quoi ?

File: src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export const CreditProgressBarSmallWithSeparator: React.FC<Props> = ({ progress }) => {
```
Comment (marineshaw) [inline src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx:12]: Je trouve le naming un peu long, tu penses qu'on pourrait pas juste mettre CreditBarWithSeparator ?

File: src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx (hunk 1 above)
Comment (Lucasbeneston) [reply src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx:10]: Comme c'est un composant avec un affichage en "dur" je trouve que ça ne sert pas a grand chose ? Surtout qu'avec des props de ce genre, on risque d'avoir des "3/2" qui cassent. Genre ça ne doit pas être trop permissif

File: src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx (hunk 2 above)
Comment (Lucasbeneston) [reply src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx:12]: Oui, le "small" n'est pas obligatoire

File: __snapshots__/features/profile/components/Header/CreditHeader/CreditHeader.native.test.tsx.native-snap [hunk 3]
```diff
@@ -1131,6 +1133,7 @@ exports[`CreditHeader Beneficiary is underage should render correctly for undera
         }
       >
         <View
+          height="normal"
           style={
```
[the remark this answers is not in the document]
Comment (Lucasbeneston) [reply __snapshots__/features/profile/components/Header/CreditHeader/CreditHeader.native.test.tsx.native-snap:1137]: Je fais au plus rapide / simple j'avoue :D

File: src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx (hunk 1 above)
Comment (marineshaw) [reply src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx]: On peut faire une condition dans le code qui retourne null si CurrentStep > TotalStep ? J'ai l'impression que dans ton code on peut aussi avoir 2/4 3/5 de manière assez simple, ça pourrait être cool si on veut réutiliser ce composant à l'avenir (je me trompe peut être ?)

Comment (Lucasbeneston) [reply src/features/profile/components/CreditInfo/CreditProgressBarSmallWithSeparator.tsx]: Oui le code est fait pour ça, mais on anticipe peut-être trop ? :D
Après si tu veux vraiment diviser en deux props je peux le faire, mais peut-être pas aujourd'hui comme il y a le problème de l'inscription / date de naissance

**Review by marineshaw — APPROVED** [review]

**Pull request merged**